src/argilla/client/sdk/users/api.py

```diff
@@ -42,7 +41,20 @@ def whoami(
     url = "/api/me"
 
     response = client.get(url)
-    return UserModel(**response)
+    # TODO(alvarobartt,frascuchon): use ONLY `httpx.Client` instead of `AuthenticatedClient` and
+    # fix mock in `tests/conftest.py` to use `httpx.Client` instead of `AuthenticatedClient`
+    if isinstance(response, httpx.Response):
+        if response.status_code == 200:
+            return Response(
+                status_code=response.status_code,
+                content=response.content,
+                headers=response.headers,
+                parsed=UserModel(**response.json()),
+            )
+        else:
+            handle_response_error(response)
+    else:
+        return UserModel(**response)
```

Comment:
I agree with this TODO, but I guess we can revert this for the moment, until we create an actual PR to refactor it, taking both scenarios into consideration is not a good thing IMO, we already had this but decided to revert in favour of the current approach until we fix the mocks

```suggestion
    return UserModel(**response)
```